### PR TITLE
Add DateTimeImmutable extension

### PR DESCRIPTION
**Pull request by pascalheidmann**
### What is the reason for this PR?

We have a extension for generating `DateTime`s but no native way for `DateTimeImmutable` even though they have clear advantages in most sitations. Therefore this PR creates an extension for `DateTimeImmutable` with the same API as for `DateTime`.

- [x] A new feature
- [ ] Fixed an issue (resolve #ID)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

Basically copy & paste from `DateTime`

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
Please read the contribution guide. We are currently not adding new features
